workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/internal/SubtreeManager.kt

```diff
@@ -114,6 +116,39 @@ internal class SubtreeManager<PropsT, StateT, OutputT>(
     key: String,
     handler: (ChildOutputT) -> WorkflowAction<PropsT, StateT, OutputT>
   ): ChildRenderingT {
+    val stagedChild = prepareStagedChild(
+      child,
+      props,
+      key,
+      handler
+    )
+    return stagedChild.render(child.asStatefulWorkflow(), props)
+  }
+
+  @Composable
+  override fun <ChildPropsT, ChildOutputT, ChildRenderingT> Rendering(
+    child: Workflow<ChildPropsT, ChildOutputT, ChildRenderingT>,
+    props: ChildPropsT,
+    key: String,
+    handler: (ChildOutputT) -> WorkflowAction<PropsT, StateT, OutputT>
+  ): ChildRenderingT {
+    val stagedChild = remember (child, props, key, handler) {
+      prepareStagedChild(
```

Comment:
I don't believe this is safe to call from composition since it modifies non-snapshot state in this class.

Reply:
given that `SideEffect` only occurs after Composition though and we need this `stagedChild` to call the Rendering waht are the options here?

I thought `remember` would effectively avoid calling this multiple times across recomposition?

Do you mean that the StagingList needs to be `State`?

Reply:
Remember will prevent calling it again on subsequent compositions, but it will still be called if the initial composition is discarded and never applied.

Reply:
I think i'm just going to have to re-write or modify `ActiveStagingList` in a Snapshot-State-friendly way

Reply:
```
  /**
   * Prepare the staged child while only modifying [children] in a SideEffect. This will ensure
   * that we do not inappropriately modify non-snapshot state.
   */
  @Composable
  private fun <ChildPropsT, ChildOutputT, ChildRenderingT> StagedChild(
    child: Workflow<ChildPropsT, ChildOutputT, ChildRenderingT>,
    props: ChildPropsT,
    key: String,
    handler: (ChildOutputT) -> WorkflowAction<PropsT, StateT, OutputT>
  ): WorkflowChildNode<*, *, *, *, *> {
    val childState = remember (child, key, props, handler) {
      children.forEachStaging {
        require(!(it.matches(child, key))) {
          "Expected keys to be unique for ${child.identifier}: key=\"$key\""
        }
      }
      mutableStateOf(children.firstActiveMatchingOrNull {
        it.matches(child, key)
      } ?: createChildNode(child, props, key, handler))
    }

    SideEffect {
      children.swapOrStage(
        predicate = {it.matches(child, key)},
        child = childState.value)
    }
    return childState.value
  }
```